packages/graph/hash_graph/postgres_migrations/V4__ontology_functions.sql

```diff
@@ -53,36 +55,46 @@ BEGIN
 
   SET CONSTRAINTS type_ids_pkey DEFERRED;
   SET CONSTRAINTS type_id_unique DEFERRED;
+  SET CONSTRAINTS type_ids_overlapping DEFERRED;
 
   RETURN QUERY
   UPDATE type_ids
   SET
+    "transaction_time" = tstzrange(now(), NULL, '[)'),
     "version" = update_ontology_id.version,
     "version_id" = _version_id,
     "updated_by_id" = update_ontology_id.updated_by_id
   WHERE type_ids.base_uri = update_ontology_id.base_uri
-    AND type_ids.version = update_ontology_id.version - 1
+    AND type_ids.transaction_time @> now()
   RETURNING type_ids.version_id, type_ids.owned_by_id;
 
   SET CONSTRAINTS type_ids_pkey IMMEDIATE;
   SET CONSTRAINTS type_id_unique IMMEDIATE;
+  SET CONSTRAINTS type_ids_overlapping IMMEDIATE;
 END $update_ontology_id$ VOLATILE LANGUAGE plpgsql;
 
 CREATE
 OR REPLACE FUNCTION "update_type_ids_trigger" () RETURNS TRIGGER AS $pga$
     BEGIN
+      IF (OLD.version != NEW.version - 1) THEN
+        RAISE EXCEPTION 'Not updating the latest id: % -> %', OLD.version, NEW.version
+        USING ERRCODE = 'invalid_parameter_value';
```

Comment:
Oh cool, does the SQL type crate provide translations for these built-in enums? (or at least I think they are enums)

This is incredibly useful for the setup we have with the SQL functions!

Reply:
Yes, all error codes are available via `SqlState` from `tokio_postgres`. Really handy!